Mathlib/Computability/AkraBazzi/AkraBazzi.lean

```diff
@@ -0,0 +1,1505 @@
+/-
+Copyright (c) 2023 Frédéric Dupuis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Frédéric Dupuis
+-/
+
+import Mathlib.Computability.AkraBazzi.GrowsPolynomially
+import Mathlib.Analysis.Calculus.MeanValue
+import Mathlib.Analysis.SpecialFunctions.Pow.Deriv
+
+/-!
+# Divide-and-conquer recurrences and the Akra-Bazzi theorem
+
+A divide-and-conquer recurrence is a function `T : ℕ → ℝ` that satisfies a recurrence relation of
+the form T(n) = ∑_{i=0}^{k-1} a_i T(r_i(n)) + g(n) for large enough n, where r_i(n) is some
+function where ‖r_i(n) - b_i n‖ ∈ o(n / (log n)^2) for all, the a_i's are some positive
```

Comment:
you have a small o here, but Wikipedia gives the claim with a big O. Which is correct?

Reply:
It can be tightened to big-O, by choosing the "smoothing function" differently and making the proof more complicated. This would have a very low payoff though, I'm not sure there is even a single application where this makes a difference, n / (log n)^2 is already overkill most of the time.